tools/lsp/properties.rs

```diff
@@ -68,15 +72,16 @@ fn source_file(element: &Element) -> Option<String> {
     element.source_file().map(|sf| sf.path().to_string_lossy().to_string())
 }
 
-fn get_element_properties<'a>(
-    element: &'a Element,
-    offset_to_position: &'a mut dyn FnMut(u32) -> lsp_types::Position,
-    group: &'a str,
+fn add_element_properties(
+    element: &Element,
+    offset_to_position: &mut dyn OffsetToPositionMapper,
+    group: &str,
     is_local_element: bool,
-) -> impl Iterator<Item = PropertyInformation> + 'a {
+    result: &mut Vec<PropertyInformation>,
```

Comment:
Just curious: Is the change of return type related to the offset to position mapper changes?